test/mutual_information_method.jl

```diff
@@ -0,0 +1,62 @@
+using GlobalSensitivity, Test, QuasiMonteCarlo
+
+function ishi_batch(X)
+    A = 7
+    B = 0.1
+    @. sin(X[1, :]) + A * sin(X[2, :])^2 + B * X[3, :]^4 * sin(X[1, :])
+end
+function ishi(X)
+    A = 7
+    B = 0.1
+    sin(X[1]) + A * sin(X[2])^2 + B * X[3]^4 * sin(X[1])
+end
+
+function linear_batch(X)
+    A = 7
+    B = 0.1
+    @. A * X[1, :] + B * X[2, :]
+end
+function linear(X)
+    A = 7
+    B = 0.1
+    A * X[1] + B * X[2]
+end
+
+lb = -ones(4) * π
+ub = ones(4) * π
+
+res1 = gsa(
+    ishi, MutualInformation(order = [0, 1, 2]), [[lb[i], ub[i]] for i in 1:4], samples = 10_000)
+
+res2 = gsa(
+    ishi_batch, MutualInformation(order = [0, 1, 2]), [[lb[i], ub[i]] for i in 1:4],
+    samples = 10_000, batch = true)
+
+res1.S1_Conf_Int
+
+@test res1.S1≈[0.1416, 0.1929, 0.1204, 0.0925] atol=1e-3
```

Comment:
These look significantly different from the usual Sobol values, especially the fourth parameter having such a large value is concerning?

Reply:
Given the the fourth parameter doesn't show up in the model at all so it should be 0 regardless of other values, or at least close to it

Reply:
If you compute the confidence bounds using more than 1 bootstrap sample, the 4th parameter is within the bounds, so this method indicates it's not sensitive. Should I add this to the docs?

Reply:
The mutual information method only has meaning when compared to its bounds . I am not sure how useful it is in the practical sense to have the bootstraping be 1 by default instead of 1000. MI values can vary a lot depending on the distributions of x and y, irrespectively of their joint distribytions.

Reply:
I've removed the second and total order indices as they didn't make sense based on what I computed and knew from the sobol indices. I now adjusted for the upper confidence bounds and reduced this method to its most basic form. In this way, maybe the other indices can be left for a later implementation.